### ISSUE TITLE
2.2.0 Cypher Bug - Simple node by legacy index query ignores filtering

### ISSUE THREAD

**Issue by Deevian** [neo4j/neo4j#4341]
Starting in Neo4j 2.2.0, we've noticed an issue with some of the legacy query results of our existing application. An example:

```
START root=node:Person('id:deev*')
WHERE root.id = 'deevian'
RETURN root
```

This query will return every `Person` matched by the `START`, completely ignoring the following filter. Indeed, the `EXPLAIN` shows no filter being executed.

<img src="http://i.imgur.com/uBe0z8q.png"/>

By adding a `MATCH (root)` clause (or changing the `START` to `root=node(*)`, or adding whatever else, really), it starts working.

```
START root=node:Person('id:deev*')
MATCH (root)
WHERE root.id = 'deevian'
RETURN root
```

<img src="http://i.imgur.com/E9IC2FA.png"/>

**Comment by cleishm** [comment]
Thanks @Deevian. I've verified the issue and have passed it along to our team.

**Comment by cleishm** [comment]
Resolved by #4346
